### ISSUE TITLE
Question: Is it possible to use middleware to prevent the Logtail sync?

### ISSUE THREAD

**Issue by fated-x** [logtail/logtail-js#9]
The scenario is this:

- I want to use a Winston log level of "info".
- I have a File transport with a log level of "info".
- I have a File transport with a log level of "error".
- I have the Logtail transport.
- I want to continue to log to both of my files, but I only want "warn" and "error" log levels to get synced to Logtail.

Is this possible without having to add a second logger just for Logtail?

**Comment by gyfis** [comment]
@fated-x Hi Stephen! You should be able to use a middleware: https://docs.logtail.com/integrations/javascript#middleware . You can check for the log level in the middleware, and return an empty array in case it's not the log level you want.

**Comment by fated-x** [comment]
Thanks for the response, @gyfis. I finally had a moment to try this out tonight and it resulted in the following error right in the @logtail package itself:
`TypeError: Cannot read properties of undefined (reading 'toISOString')`

**3 Experiments**
1. I tried returning an empty object (`{}`) and received the same error.
2. I tried returning `null` and received a pretty similar error:
`TypeError: Cannot read properties of null (reading 'dt')`
3. I tried returning `{ dt: new Date() }` since `dt` appeared to be very much required by the middleware. The errors went away but "info" level logs started appearing in Logtail with no data as only a date object was returned from the code.

I am attempting to prevent "info" logs from getting uploaded at all. Any help would be greatly appreciated. Thanks!

And here is the logtail middleware I am using, for reference:
```
const logtail = new Logtail(process.env.LOGTAIL_NIGHTLY_UPDATE_SOURCE_TOKEN);

async function filterOutLog(log) {
  return ['error', 'warn'].includes(log.level) ? log : [];
}

logtail.use(filterOutLog);
```

**Comment by adikus** [comment]
@fated-x Sorry about the delay on this issue! I've created a PR for this that should allow you to do what you need. I'm hoping we can get it merged and released fairly soon.

**Comment by adikus** [comment]
@fated-x I released 0.1.12 which should allow you to filter logs in your middleware now - you can return `null` from the middleware to get the log skipped. Using your example, it would look something like this:
```
const logtail = new Logtail(process.env.LOGTAIL_NIGHTLY_UPDATE_SOURCE_TOKEN);

async function filterOutLog(log) {
  return ['error', 'warn'].includes(log.level) ? log : null;
}

logtail.use(filterOutLog);
```
